packages/data-stores/src/domain-suggestions/resolvers.ts

```diff
@@ -6,9 +6,24 @@ import { stringify } from 'qs';
 /**
  * Internal dependencies
  */
-import { receiveCategories, receiveDomainSuggestions } from './actions';
+import { receiveCategories, receiveDomainSuggestions, receiveDomainAvailability } from './actions';
 import { fetchAndParse, wpcomRequest } from '../wpcom-request-controls';
 import type { Selectors } from './selectors';
+import type { TailParameters } from '../mapped-types';
+
+export const isAvailable = function* isAvailable(
+	domainName: TailParameters< Selectors[ 'isAvailable' ] >[ 0 ]
```

Comment:
It's a bit weird to use `TailParameters<…>[0]` and `Parameters<…>[1]` (effectively the same) in the same file, but it gets the job done 🙂